src/geometry/WorkingGeometry.ts

```diff
@@ -143,22 +148,60 @@ export class WorkingGeometry {
     /**
      * Scale by pulling a point to a new point.
      *
-     * @param {vec3} pullPoint: point that scales to destination_point after transformation.
-     * @param {vec3} destinationPoint: point that is the result of pull_point after transformation.
+     * @param {vec3} pullPoint: point that scales to destinationPoint after transformation.
+     * @param {vec3} destinationPoint: point that is the result of pullPoint after transformation.
      * @param {vec3} holdPoint: point to scale from, default to true origin.
      */
     public scale(pullPoint: vec3, destinationPoint: vec3, holdPoint: vec3 = vec3.create()) {
-        const destinationVector: vec3 = vec3.sub(vec3.create(), destinationPoint, holdPoint);
-        const pullVector: vec3 = vec3.sub(vec3.create(), pullPoint, holdPoint);
-        const scalingVector: vec3 = this.getScalingVector(destinationVector, pullVector);
-        const scalingMatrix: mat4 = mat4.fromRotationTranslationScaleOrigin(
+        const moveHoldToOrigin = mat4.translate(
             mat4.create(),
-            quat.create(),
-            vec3.create(),
-            scalingVector,
-            holdPoint
+            mat4.create(),
+            vec3.negate(vec3.create(), holdPoint)
         );
-        this.transform(scalingMatrix);
+
+        const translatedPullPoint = vec3.sub(vec3.create(), pullPoint, holdPoint);
+        const translatedDestinationPoint = vec3.sub(vec3.create(), destinationPoint, holdPoint);
+
+        const rotPullToXAxis = mat4.fromQuat(
+            mat4.create(),
+            quat.rotationTo(
+                quat.create(),
+                vec3.normalize(vec3.create(), translatedPullPoint),
+                vec3.fromValues(1, 0, 0)
+            )
+        );
+
+        const scalePullToDestOnXAxis = mat4.scale(
+            mat4.create(),
+            mat4.create(),
+            vec3.fromValues(
+                vec3.len(translatedDestinationPoint) / vec3.len(translatedPullPoint),
+                1,
+                1
+            )
+        );
+
+        const rotXAxisToDestAxis = mat4.fromQuat(
+            mat4.create(),
+            quat.rotationTo(
+                quat.create(),
+                vec3.fromValues(1, 0, 0),
+                vec3.normalize(vec3.create(), translatedDestinationPoint)
+            )
+        );
+
+        const moveOriginToHold = mat4.translate(mat4.create(), mat4.create(), holdPoint);
+
+        // Assemble all the matrices!
+        // Note that matrix is modified after each mul call, but the linter can't figure that out,
+        // so the const is only there to appease the linter
+        const matrix = moveOriginToHold;
```

Comment:
`const` only means a const reference, and since we're modifying the contents of the variable without changing the reference, `const` still makes sense.

Reply:
Got it. I'll remove that comment.